### PR TITLE
INGM-276 Integrate comms refactor

### PR DESCRIPTION
**Pull request by MartinALL91**
### Description

Integrate ingenialink's communication refactor.

Fixes # INGM-276

### Type of changes
- Remove IPB references.
- Create a utils module.
- Remove connect_servo_ecat and connect_servo_ecat_interface_ip functions.
- Create connect_servo_eoe_service and connect_servo_eoe_service_interface_index functions.
- Update Jenkinsfile to skip the EtherCAT stages.
- Update CHANGELOG.

### Tests
- Run Jenkins job on branch `test-INGM-276`

### CODE REVIEW

**Review by WoodElv — APPROVED** [review]

File: ingeniamotion/communication.py [hunk 1]
```diff
@@ -128,137 +109,73 @@ def __servo_connect(
...
-            reconnection_retries : Number of reconnection retried before declaring
-                a connected or disconnected stated.
-            reconnection_timeout : Time in ms of the reconnection timeout.
             servo_status_listener : Toggle the listener of the servo for
                 its status, errors, faults, etc.
             net_status_listener : Toggle the listener of the network
                 status, connection and disconnection.
 
         Raises:
-            FileNotFoundError: If the dict file doesn't exist.
```
Comment (GullinBustin) [inline ingeniamotion/communication.py:189]: Why did you remove the FileNotFoundError?

File: ingeniamotion/utils.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+from typing import Union
+
+from ingenialink.ethernet.register import EthernetRegister
+from ingenialink.canopen.register import CanopenRegister
+
+
+def map_register_address(register: Union[EthernetRegister, CanopenRegister]) -> int:
```
Comment (GullinBustin) [inline ingeniamotion/utils.py:7]: Don't create an utils.py file, it is an antipattern https://www.yanglinzhao.com/posts/utils-antipattern/.
And create a test for this function.

**Review by GullinBustin — CHANGES_REQUESTED** [review]

File: ingeniamotion/communication.py (hunk 1 above)
Comment (MartinALL91) [reply ingeniamotion/communication.py:189]: I did not notice. I'll add it back.

File: ingeniamotion/communication.py [hunk 3]
```diff
@@ -128,137 +109,73 @@ def __servo_connect(
...
+            port,
             servo_status_listener=servo_status_listener,
             net_status_listener=net_status_listener,
         )
         servo.slave = slave
-
         self.mc.servos[alias] = servo
         self.mc.servo_net[alias] = alias
 
-    def connect_servo_ecat_interface_ip(
```
[the remark this answers is not in the document]
Comment (MartinALL91) [reply ingeniamotion/communication.py:217]: My bad, I'll add it back.

File: ingeniamotion/utils/monitoring.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
+from typing import Union
+
+from ingenialink.ethernet.register import EthernetRegister
+from ingenialink.canopen.register import CanopenRegister
+
+
+def map_register_address(register: Union[EthernetRegister, CanopenRegister]) -> int:
```
Comment (pzinemanas) [inline ingeniamotion/utils/monitoring.py:7]: This function could be a static method of Monitoring or Capture. Right? Do we need a utils folder for this?

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,13 @@
 # Changelog
 
+## [Unreleased]
+### Added
+- connect_servo_eoe_service and connect_servo_eoe_service_interface_index functions are added.
+- The utils module.
```
Comment (pzinemanas) [inline CHANGELOG.md:6]: I'd rather avoid having a utils module.

**Review by pzinemanas — CHANGES_REQUESTED** [review]

File: ingeniamotion/utils/monitoring.py (hunk 4 above)
Comment (MartinALL91) [reply ingeniamotion/utils/monitoring.py:7]: This method is also used in disturbance, I don't like the idea of using a static method from Monitoring there. I could not think of a better place to put this function besides in a utils module.

File: ingeniamotion/communication.py [hunk 6]
```diff
@@ -128,133 +109,112 @@ def __servo_connect(
...
-            reconnection_retries : Number of reconnection retried before declaring
-                a connected or disconnected stated.
-            reconnection_timeout : Time in ms of the reconnection timeout.
             servo_status_listener : Toggle the listener of the servo for
                 its status, errors, faults, etc.
             net_status_listener : Toggle the listener of the network
                 status, connection and disconnection.
 
         Raises:
+            TypeError: If the dict_path argument is missing.
```
Comment (GullinBustin) [inline ingeniamotion/communication.py:161]: Why did you add this? dict_path is not optional, so of course an error will be raised if the argument is missing. I think it is redundant.

**Review by GullinBustin — CHANGES_REQUESTED** [review]

File: ingeniamotion/communication.py (hunk 6 above)
Comment (MartinALL91) [reply ingeniamotion/communication.py:161]: My bad, I'll remove it.

File: ingeniamotion/communication.py [hunk 7]
```diff
@@ -79,10 +68,8 @@ def connect_servo_ethernet(
         ip: str,
         dict_path: Optional[str] = None,
```
[the remark this answers is not in the document]
Comment (MartinALL91) [reply ingeniamotion/communication.py:69]: Ok, it was an optional argument but an exception was raised if the argument had the default value haha. I'll fix it.

**Review by pzinemanas — APPROVED** [review]

File: ingeniamotion/utils/monitoring.py (hunk 4 above)
Comment (MartinALL91) [reply ingeniamotion/utils/monitoring.py]: We have a circular import error when moving the `map_register_address `function to Capture. I think I'll revert back to using a utils module.

**Review by GullinBustin — APPROVED** [review]
